src/libYARP_sig/include/yarp/sig/PointCloudTypes.h

```diff
@@ -0,0 +1,945 @@
+            PCL_POINT_XYZ              =   (PC_XYZ_DATA),
+            PCL_POINT_XYZ_RGBA         =   (PC_XYZ_DATA | PC_RGBA_DATA | PC_PADDING3),
+            PCL_POINT_XYZ_I            =   (PC_XYZ_DATA | PC_INTENSITY_DATA),
+            PCL_INTEREST_POINT_XYZ     =   (PC_XYZ_DATA | PC_INTEREST_DATA),
+            PCL_NORMAL                 =   (PC_NORMAL_DATA | PC_CURVATURE_DATA | PC_PADDING3),
+            PCL_POINT_XYZ_NORMAL       =   (PC_XYZ_DATA | PC_NORMAL_DATA | PC_CURVATURE_DATA | PC_PADDING3),
+            PCL_POINT_XYZ_NORMAL_RGBA  =   (PC_XYZ_DATA | PC_RGBA_DATA | PC_NORMAL_DATA | PC_CURVATURE_DATA | PC_PADDING2),   // Actually PCL has PointXYZRGBNormal, not RGBA, but downgrade from rgba to rgb can be done while casting
+            PCL_POINT_XYZ_I_NORMAL     =   (PC_XYZ_DATA | PC_INTENSITY_DATA | PC_NORMAL_DATA | PC_CURVATURE_DATA),
+            PCL_POINT_XYZ_RANGE        =   (PC_XYZ_DATA | PC_RANGE_DATA),
+            PCL_POINT_XYZ_VIEWPOINT    =   (PC_XYZ_DATA | PC_VIEWPOINT_DATA),
+            PCL_MOMENT_INVARIANTS      =   (PC_MOMENT_INV_DATA),
+            PCL_PRINCIPAL_RADII_RSD    =   (PC_RADII_RSD_DATA),
+            PCL_BOUNDARY               =   (PC_BOUNDARY_DATA),
+            PCL_PRINCIPAL_CURVATURES   =   (PC_PRINCIPAL_CURVATURE_DATA),
+            PCL_PFH_SIGNAT_125         =   (PC_PFH_SIGNAT_125_DATA),
+            PCL_FPFH_SIGNAT_33         =   (PC_FPFH_SIGNAT_33_DATA),
+            PCL_VFH_SIGNAT_308         =   (PC_VFH_SIGNAT_308_DATA),
+            PCL_NARF_36                =   (PC_NARF_36_DATA),
+            PCL_POINT2D_BORDER         =   (PC_XY_DATA | PC_BORDER_DATA),
+            PCL_INTENSITY_GRADIENT     =   (PC_INTENSITY_GRAD_DATA),
+            PCL_PC_HISTOGRAM_N         =   (PC_HISTOGRAM_DATA),
+            PCL_POINT_XYZ_SCALE        =   (PC_XYZ_DATA | PC_SCALE_DATA),
+            PCL_POINT_XYZ_SURFEL       =   (PC_XYZ_DATA | PC_RGBA_DATA | PC_NORMAL_DATA | PC_RADIUS_DATA | PC_CONFIDENCE_DATA)
+        };
+
+        // Defined as in PCL pointTypes.h file for better compatibility
+        enum YarpBorderTraitEnum
+        {
+            BORDER_TRAIT__OBSTACLE_BORDER,
+            BORDER_TRAIT__SHADOW_BORDER,
+            BORDER_TRAIT__VEIL_POINT,
+            BORDER_TRAIT__SHADOW_BORDER_TOP,
+            BORDER_TRAIT__SHADOW_BORDER_RIGHT,
+            BORDER_TRAIT__SHADOW_BORDER_BOTTOM,
+            BORDER_TRAIT__SHADOW_BORDER_LEFT,
+            BORDER_TRAIT__OBSTACLE_BORDER_TOP,
+            BORDER_TRAIT__OBSTACLE_BORDER_RIGHT,
+            BORDER_TRAIT__OBSTACLE_BORDER_BOTTOM,
+            BORDER_TRAIT__OBSTACLE_BORDER_LEFT,
+            BORDER_TRAIT__VEIL_POINT_TOP,
+            BORDER_TRAIT__VEIL_POINT_RIGHT,
+            BORDER_TRAIT__VEIL_POINT_BOTTOM,
+            BORDER_TRAIT__VEIL_POINT_LEFT,
+        };
+
+        // Definition of single fields data structures
+        YARP_BEGIN_PACK
+        struct XYData
+        {
+            union
+            {
+                float _xy[2];
```

Comment:
Vector and Image doesn't check the endianess:
>
```/**
* \ingroup sig_class
*
* A Base class for a VectorOf<T>, provide default implementation for
* read/write methods. Warning: the current implementation assumes the same
* representation for data type (endianess).
*/ 
```
Since we are using internally a `yarp::sig::VectorOf` the PointCloud is affected by the same problem

Reply:
Ok, let's forget about this... The NetworkHeader comment is important though, it's not just endianness, also size of bool can change on different architectures